test/testlib/nuodb_database_utilities.go

```diff
@@ -786,3 +789,77 @@ func SnapshotVolume(t *testing.T, namespaceName string, pvcName string, snapName
 		return error == nil && strings.TrimSpace(out) == "'true'"
 	}, 30*time.Second)
 }
+
+func getSnapshotName(t *testing.T, tmpl *template.Template, backupId, volumeType string) string {
+	var buf bytes.Buffer
+	err := tmpl.Execute(&buf, &map[string]string{
+		"backupId":   backupId,
+		"volumeType": volumeType,
+	})
+	require.NoError(t, err)
+	return buf.String()
+}
+
+func getArchiveSnapshotName(t *testing.T, tmpl *template.Template, backupId string) string {
+	return getSnapshotName(t, tmpl, backupId, "archive")
+}
+
+func getJournalSnapshotName(t *testing.T, tmpl *template.Template, backupId string) string {
+	return getSnapshotName(t, tmpl, backupId, "journal")
+}
+
+type BackupHookResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+}
+
+func GetBackupHookResponse(t *testing.T, namespaceName, smPod, resourcePath string) *BackupHookResponse {
+	// Issue request to backup hook server. The nuodb image has curl, so use
+	// it in the SM container to issue HTTP request against the backup hook
+	// server running in a sidecar container on the same pod.
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--",
+		"curl", "-s", "-X", "POST", "http://0.0.0.0:8000/"+resourcePath)
```

Comment:
In the real scenario, I presume that we need to use `http://<POD_IP>:8080` URL to invoke the backup hooks?

Reply:
Yes, it should work with `http://<pod IP>:8000` assuming that you have access to the internal network. I didn't exercise it that way because I don't think the pod IPs are routable in the test.